test/any/web_catalog/api-extractor-test.es6.js

```diff
@@ -0,0 +1,331 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+(function() {
+  let objectGraph = global.objectGraph;
+  let extractor = com.web.catalog.apiExtractor.create({});
+  const types = {
+    'boolean': 2,
+    'exception': 7,
+    'null': 6,
+    'number': 3,
+    'string': 4,
+    'symbol': 5,
+    'undefined': 1,
+  };
+
+  describe('Method getObjectProperties', function() {
+    let og = objectGraph.fromJSON({
+      'data': {
+        '10000': {
+          'constantNumber': 3,
+          'constantObject': 6,
+          'nonOwnProperty': 3,
+          'property': 3,
+          'object': 7,
+          'buildInProperty': 3,
+        },
+      },
+      'metadata': {
+        '10000': {
+          'constantNumber': {
+            writable: 0,
+          },
+          'constantObject': {
+            writable: 0,
+          },
+          'property': {
+            writable: 1,
+          },
+          'object': {
+            writable: 1,
+          },
+          'buildInProperty': {
+            writable: 1,
+          },
+        },
+      },
+      types,
+    });
+    let properties = extractor.getObjectProperties_(og, 10000);
+    extractor.buildInObjectProperties = ['buildInProperty'];
+    it('captures non-constant properties', function() {
+      expect(properties).toContain('property');
+    });
+    it('captures object regardless if it is constant or not', function() {
+      expect(properties).toContain('object');
+    });
+    it('excludes constant property', function() {
+      expect(properties).not.toContain('constantNumber');
+    });
+    it('excludes non-own-property', function() {
+      expect(properties).not.toContain('nonOwnProperty');
+    });
+    it('does not filter build-in properties', function() {
+      expect(properties).toContain('buildInProperty');
+    });
+    // Turn on retainConstantMembers flag.
```

Comment:
Here and elsewhere: Try to have your comments answer "why?" more than "what?" (if the "what" isn't clear, then maybe we need to rename APIs or restructure the code to be clearer).

E.g., for this one: `// Need to test retaining constant members case separately.`